packages/credit-protocol/index.ts

```diff
@@ -418,15 +418,31 @@ export default class CreditProtocol {
     return config.gasPrice
   }
 
-  async getTxCost(currency: string, gasNeeded: number) {
+  async getTransactionCosts(currency: string, gasNeeded: number) {
     try {
       const gasPrice = await this.getGasPrice()
       const rate = await this.getEthExchange(currency)
+      const weiCost = gasPrice * gasNeeded
+      const ethCost = weiCost / WEI_PER_ETH
```

Comment:
I'd be careful with these numbers. They can easily exceed the bounds of what JS can reliably represent. Instead, use BigNumber or BN.js to perform operations on and to represent potentially huge numbers.